docs/specifications/tests/Methods.md

```diff
@@ -1,69 +1,590 @@
+
+3. Get IP addresses for any [out-of-bailiwick] name server
+   names in *Names* by running [method-get-oob-ips] with *Names*
+   as input.
+
+4. Merge the set returned with *Name Servers*.
+
+5. Return *Name Servers*.
+
+### Outcome(s)
+
+Outcome is the set of name servers, where each unique name server name
+links to a possibly empty set of its IP addresses.
+
+### Special procedural requirements
+
+None.
+
+### Dependencies
+
+Test Case [BASIC01] must have been run.
+
+-------------------------------------------------------------
+
+## Method: Get zone NS IP addresses
+
+### Method identifier
+**method-get-zone-ns-ips**
+
+### Objective
+
+Obtain the IP addresses of the name servers, as extracted from 
+the NS records of apex of the child zone.
+
+This is an external method to be used by test cases.
+
+### Inputs
+
+* "Child Zone" - The name of the child zone.
+
+### Prerequisite
+
+* As specified in [method-get-delegation].
+
+### Ordered description of steps to be taken to execute the method
+
+1. Get the name servers set ("Name Servers") where each unique name 
+   server name is linked to a set of its possibly empty set of IP 
+   addresses by running [method-get-delegation].
+
+2. Extract the set of name server names ("Names") from 
+   *Name Servers*.
+
+3. Get IP addresses for any [out-of-bailiwick] name server
+   names in *Names* by running [method-get-oob-ips] with *Names*
+   as input.
+
+4. Merge the set returned with *Name Servers*.
+
+5. Extract the IP addresses from *Name Servers* and create a
+   set of unique IP addresses.
+
+5. Return set of IP addresses.
+
+### Outcome(s)
+
+Outcome is a possibly empty set of IP addresses.
+
+### Special procedural requirements
+
+None.
+
+### Dependencies
+
+Test Case [BASIC01] must have been run.
+
+-------------------------------------------------------------
+
+## Method: Get in-bailiwick address records in zone
+
+### Method identifier
+**method-get-ib-addr-in-zone**
+
+### Objective
+
+From the child zone, obtain the address records matching the
+in-bailiwick name server names found in the zone itself.
+
+This is an internal method to be used by other methods.
+
+### Inputs
+
+* "Child Zone" - The name of the child zone.
+
+### Prerequisite
+
+* As specified in [method-get-del-ns-ips].
+
+### Ordered description of steps to be taken to execute the method
+
+1. Using [method-get-del-ns-ips] get the IP addresses to the
+   name servers ("Name Server IPs").
+
+2. Using [method-get-zone-ns-names] get the names of the name
+   servers according to *Child Zone* itself 
+   ("Child Zone name server name").
+
+3. If none of the *Child Zone name server name* is an
+   [in-bailiwick] name server name, exit this method with an empty
+   set.
+
+4. Create a set name servers ("Name Servers") where each unique 
+   [in-bailiwick] name server name is linked to an empty set  
+   of IP addresses. 
+
+5. For each *Child Zone name server name* that is an [in-bailiwick]
+   name server:
```

Comment:
Qualifications similar to "that is ..." is a common theme across these specifications. Oftentimes they're intended as helpful clarifications to the reader. At other times, like in this case, they're explicit instructions to filter out a subset of the named data. It's very taxing on the reader to determine whether the author is being nice of if he's instructing me to perform some operation that makes a real difference.

To remove any doubt of this kind, I suggest a threefold solution:
1. Make filtering operations into steps of their own, introducing a new name for the result if necessary.
2. Move the helpful clarification to the output specification of the called method (it will be easily reachable using the method link).
3. Put the method Outcome(s) headings close to the top of their respective method specification.